### ISSUE TITLE
decode_content removes header

### ISSUE THREAD

**Issue by phmLabs** [guzzle/guzzle#1397]
When using the decode_content option the "Content-Encoding" header is removed. 

Is it possible to get the header to check if the content was gzipped?

**Comment by mtdowling** [comment]
Are you using the curl or streamhandler? Do you know if the behavior between the two is different? If curl isn't removing the header, but the StreamHandler is, then I'd be open to stopping this behavior in the StreamHandler.

**Comment by phmLabs** [comment]
It seems to be this line:

```php
unset($headers[$normalizedKeys['content-encoding']]);
```

in 

```
https://github.com/guzzle/guzzle/blob/master/src/Handler/StreamHandler.php
```

**Comment by jeskew** [comment]
The cURL handler [will also remove the Content-Encoding header when `decode_content` is set to true](https://github.com/guzzle/guzzle/blob/739b9c8378c462943c42fee5353453ab4164ed8e/src/Handler/EasyHandle.php#L59).

**Comment by mtdowling** [comment]
It should be noted that others have requested the opposite of what is being proposed in this issue: https://github.com/guzzle/guzzle/issues/1075

**Comment by jeskew** [comment]
This was addressed in #1409. Guzzle will add headers containing the original content encoding (x-encoded-content-encoding) and original content length (x-encoded-content-length) and continue removing the content-encoding header and updating the content-length header with the length of the decoded body.
